services/jupyterhub/jupyterhub_config.py

```diff
@@ -0,0 +1,710 @@
+
+## The cookie secret to use to encrypt cookies.
+#
+#  Loaded from the JPY_COOKIE_SECRET env variable by default.
+#
+#  Should be exactly 256 bits (32 bytes).
+#c.JupyterHub.cookie_secret = b''
+
+## File in which to store the cookie secret.
+#c.JupyterHub.cookie_secret_file = 'jupyterhub_cookie_secret'
+
+## The location of jupyterhub data files (e.g. /usr/local/share/jupyter/hub)
+#c.JupyterHub.data_files_path = '/srv/jupyterhub'
+
+## Include any kwargs to pass to the database connection. See
+#  sqlalchemy.create_engine for details.
+#c.JupyterHub.db_kwargs = {}
+
+## url for the database. e.g. `sqlite:///jupyterhub.sqlite`
+#c.JupyterHub.db_url = 'sqlite:///jupyterhub.sqlite'
+
+## log all database transactions. This has A LOT of output
+#c.JupyterHub.debug_db = False
+
+## The ip or hostname for proxies and spawners to use for connecting to the Hub.
+#
+#  Use when the bind address (`hub_ip`) is 0.0.0.0 or otherwise different from
+#  the connect address.
+#
+#  Default: when `hub_ip` is 0.0.0.0, use `socket.gethostname()`, otherwise use
+#  `hub_ip`.
+#
+#  .. versionadded:: 0.8
+#c.JupyterHub.hub_connect_ip = ''
+
+## The port for proxies & spawners to connect to the hub on.
+#
+#  Used alongside `hub_connect_ip`
+#
+#  .. versionadded:: 0.8
+#c.JupyterHub.hub_connect_port = 0
+
+## The ip address for the Hub process to *bind* to.
+#
+#  See `hub_connect_ip` for cases where the bind and connect address should
+#  differ.
+c.JupyterHub.hub_ip = 'jupyterhub'
+
+## The port for the Hub process
+c.JupyterHub.hub_port = 8080
+
+## The public facing ip of the whole application (the proxy)
+#c.JupyterHub.ip = ''
+
+# c.JupyterHub.api_tokens = {
+#     '0fb7b7f5f2ed069547dcef6e32de21173f987c95697f3bc2c548a5b1334706dc': 'admin'
+# }
+
+## Dict of token:servicename to be loaded into the database.
+#
+#  Allows ahead-of-time generation of API tokens for use by externally managed
+#  services.
+c.JupyterHub.service_tokens = {'renga_secret1234': 'renga'}
+
+## List of service specification dictionaries.
+#
+#  A service
+#
+#  For instance::
+#
+#      services = [
+#          {
+#              'name': 'cull_idle',
+#              'command': ['/path/to/cull_idle_servers.py'],
+#          },
+#          {
+#              'name': 'formgrader',
+#              'url': 'http://127.0.0.1:1234',
+#              'api_token': 'super-secret',
+#              'environment':
+#          }
+#      ]
+
+env = os.environ.copy()
+env['FLASK_APP'] = 'project_service.py'
+env['FLASK_DEBUG'] = '1'
+env['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
+
+c.JupyterHub.services = [{
+    'name': 'projects',
+    'command': ['flask', 'run', '-p', '8000'],
+    'url': 'http://localhost:8000',
+    'environment': env,
+    'admin': True
+}]
+
+## The class to use for spawning single-user servers.
+#
+#  Should be a subclass of Spawner.
+# Spawn single-user servers as Docker containers
+from dockerspawner import DockerSpawner
+
+
+class RengaSpawner(DockerSpawner):
```

Comment:
In case you want to launch custom images, cf binderSpawner -> https://github.com/jupyterhub/binderhub/blob/master/helm-chart/binderhub/values.yaml#L106-L128